core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -285,8 +383,64 @@ object Uri {
   }
 
   type Path = String
+
+  object Path {
```

Comment:
All of these will get a lot more sophisticated on the new model in main, and the merge is going to suck.  This will be one of the first really interesting merges on cats-effect-3-dotty.